native-wheels-frontend/app/(tabs)/available-cars.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+import {
+  StyleSheet,
+  Image,
+  View,
+  Text,
+  ScrollView,
+  TouchableOpacity,
+} from "react-native";
+import axios from "axios";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { useNavigation } from "@react-navigation/native";
+
+interface Car {
+  _id: string;
+  make: string;
+  model: string;
+  year: number;
+  available: boolean;
+  image: string;
+  cylinders: number;
+  horsePower: number;
+  imageUrl: string;
+}
+
+export default function AvailableCars() {
+  const navigation = useNavigation();
+  const [cars, setCars] = useState<Car[]>([]);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchCars = async () => {
+      try {
+        const token = await AsyncStorage.getItem("token");
+        const response = await axios.get<Car[]>(
+          "http://localhost:8080/api/cars",
+          {
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+          }
+        );
```

Comment:
I suggest separating the API call and type definition of `Car` from the page, similar to how the auth calls are structured.

By importing the `axios/instance.ts`, both the auth header and base URL are automatically set, simplifying API calls. For reference, check the `getUserInfo()` function in `axios/auth/api.ts` and the related `axios/auth/types.ts`.

Reply:
I agree, much better. I separated the api and type into a cars folder under `axios/cars`. I also separated the Car banner into a separate component in `components/cars` that maybe @Olsedyr can reuse for showing booked cars on the profile page.